room/compiler-processing/src/main/java/androidx/room/compiler/processing/javac/kotlin/KotlinClassMetadataUtils.kt

```diff
@@ -188,24 +190,34 @@ private class ConstructorReader(val result: MutableList<KmConstructor>) : KmClas
     }
 }
 
-internal fun KotlinClassMetadata.Class.isObject(): Boolean = ObjectReader().let {
-    this@isObject.accept(it)
-    it.isObject
+internal class KotlinMetadataClassFlags(val classMetadata: KotlinClassMetadata.Class) {
+
+    private val flags: Flags by lazy {
```

Comment:
Took a different approach with the flag access and caching here, lmk what you think

Reply:
lgtm. cc: @danysantiago

Reply:
Since these are just class level flags I would avoid using `toKmClass()` since that parses *everything* (visits all properties and functions), instead you can get ahold of the flags with a small visitor:
```
private val flags: Flags by lazy {
    var theFlags: Flags = 0
    classMetadata.accept(object : KmClassVisitor() {
        override fun visit(flags: Flags, name: ClassName) {
            theFlags = flags
            super.visit(flags, name)
        }
    })
    return@lazy theFlags
}
```

Reply:
good tip, thanks